File: assignment 4/python/program2.py

N = int(input("Enter number of students: "))

max_ratio = 0.0  # To store the highest ratio

# Loop through each student
for i in range(N):
    print(f"Student {i + 1}:")
    height = float(input("  Enter height: "))
    weight = float(input("  Enter weight: "))

    if height != 0:
        ratio = weight / height
        if ratio > max_ratio:
            max_ratio = ratio

# Show the result
print(f"\nHighest weight/height ratio is: {max_ratio:.2f}")

File: assignment 4/C programming/program2.c
//Write a program to collect height and weight of N students. Find the highest weight/height ratio
#include <stdio.h>

struct Student {
    float height;
    float weight;
};

int main() {
    int N;
    printf("Enter the number of students: ");
    scanf("%d", &N);

    struct Student students[N];
    float maxRatio = 0.0;

    for (int i = 0; i < N; i++) {
        printf("Enter height and weight for student %d (format: height weight): ", i + 1);
        scanf("%f %f", &students[i].height, &students[i].weight);

        // Calculate weight/height ratio
        if (students[i].height != 0) {
            float ratio = students[i].weight / students[i].height;
            if (ratio > maxRatio) {
                maxRatio = ratio;
            }
        }
    }

    printf("Highest weight/height ratio: %.2f\n", maxRatio);
    return 0;
}